### PR TITLE
perf(qwen3): add crossrow QMV verifier

### PR DESCRIPTION
**Pull request by dusterbloom**
## Summary

Port the smallest validated Qwen3.8-27B performance path from the nightly work into current `upstream/main`.

- Add the crossrow Metal QMV verifier for affine 4-bit, group-64 weights.
- Enable only validated M2/M3/M5/M6/M8/M9 shapes; M4/M7 remain on stock QMV.
- Fall back to stock QMV on opt-out, unsupported shapes, reshape failures, or kernel errors.
- Pin the required MLX 0.31 fork in `Cargo.toml`/`Cargo.lock`.
- Keep Qwen3-Next weight loading chunked/evaluated for the measured path.

## Validation

- `cargo fmt --all -- --check`
- `cargo check -p higgs-models`
- `cargo test -p higgs-models --lib crossrow_qmv -- --nocapture --test-threads=1` (5 passed including M4/M7 fallback contract)
- `cargo build --release`
- `omen diff -t upstream/main`: 5 files, 668 additions, 7 deletions; medium risk due to the single kernel implementation.

The benchmark result motivating this port was Qwen3.8-27B depth-5 Low Power Mode validation at 7.39 tok/s median versus 5.85 tok/s stock, with the exact verifier trajectory preserved.

### CODE REVIEW

**Comment by dusterbloom** [comment]
Found a live correctness bug in this crossrow path while debugging repetition loops on the Escha W2 checkpoint — fixed in 201afa97d.

**Bug:** the Metal kernel hard-codes `reinterpret_cast<const device bfloat16_t*>(x)`, but nothing gates the input dtype. When a QLinear with affine-4/g64 weights receives an **fp32** activation (here: the GDN `in_proj_ba` projection during short suffix prefills, m ∈ {2,3,5,6,8,9}), the buffer is misread — halved row stride plus garbage odd 16-bit words — producing corrupt values and NaN bit patterns. Downstream this surfaced as all-248320-logits NaN (verified with a fail-fast finiteness tripwire in `sample()`) and stable one-token doom loops.

**Evidence (nightly + serving branch):** same prompt, same build — crossrow enabled → `!!!!!…` repetition loop / NaN logits; `HIGGS_CROSSROW_QMV=0` → clean; with the dtype gate → clean and **token-identical** to the disabled run.

**Fix:** eligibility now requires `x.dtype() == Dtype::Bfloat16`; everything else falls back to stock quantized matmul, which is the path this PR already documents as the correctness baseline. Build + crossrow bit-exact tests pass locally. (This branch predates the Escha loader, so the live repro itself can't run here — it was verified on the branches that carry the serving path.)

The kernel could later be taught to handle f32 natively; until then the gate keeps the fast path bit-exact and correctness-neutral.
